trigger/host_spawn.go

```diff
@@ -272,3 +275,120 @@ func (t *spawnHostStateChangeTriggers) emailPayload(action, result, hostID, url,
 		PlainTextContents: false,
 	}
 }
+
+type spawnHostSetupScriptTriggers struct {
+	hostBase
+}
+
+func makeSpawnHostSetupScriptTriggers() eventHandler {
+	t := &spawnHostSetupScriptTriggers{}
+	t.triggers = map[string]trigger{
+		event.TriggerOutcome: t.spawnHostSetupScriptOutcome,
+	}
+	return t
+}
+
+func (t *spawnHostSetupScriptTriggers) spawnHostSetupScriptOutcome(sub *event.Subscription) (*notification.Notification, error) {
+	if !t.host.UserHost {
+		return nil, nil
+	}
+	payload, err := t.makePayload(sub)
+	if err != nil {
+		return nil, errors.Wrap(err, "making payload")
+	}
+	return notification.New(t.event.ID, sub.Trigger, &sub.Subscriber, payload)
+}
+
+func (t *spawnHostSetupScriptTriggers) makePayload(sub *event.Subscription) (interface{}, error) {
+	if t.event.EventType != event.EventHostScriptExecuteFailed && t.event.EventType != event.EventHostScriptExecuted {
```

Comment:
nit: L303-313 can be simplified down slightly with a switch:
```go
var result string
switch t.event.EventType {
    case event.EventHostScriptExecuted:
        // Success logic...
    case event.EventHostScriptExecuteFailed:
        // Error logic...
    default:
        return nil, errors.Errorf("unexpected event type '%s'", t.event.EventType)
}
```

Reply:
Updated